### PR TITLE
Add function to limit permissions to selected eic function

### PR DESCRIPTION
**Pull request by FirestarJes**

### CODE REVIEW

File: libs/dh/admin/data-access-api/src/lib/dh-admin-create-user-role-management-data-access-api.store.ts [hunk 1]
```diff
@@ -57,16 +54,16 @@ export class DhAdminCreateUserRoleManagementDataAccessApiStore extends Component
     super(initialState);
   }
 
-  private readonly getSelectablePermissions = this.effect(
-    (trigger$: Observable<void>) =>
+  public readonly getSelectablePermissions = this.effect(
+    (trigger$: Observable<EicFunction>) =>
       trigger$.pipe(
-        switchMap(() =>
+        switchMap((x) =>
```
Comment (mimse) [inline libs/dh/admin/data-access-api/src/lib/dh-admin-create-user-role-management-data-access-api.store.ts:60]: Suggestion: use variable name that gives context

File: libs/dh/shared/data-access-msw/src/lib/mocks/data/marketParticipantUserRolePermissions.json [hunk 2]
```diff
@@ -1,37 +1,32 @@
 [
   {
-    "id": "3ec41d91-fc6d-4364-ade6-b85576a91d04",
-    "name": "Test role 1",
-    "status": "Active",
-    "eicFunction": "MeteredDataAdministrator",
-    "permissions": [1, 2, 3]
+    "id": "1",
```
Comment (dzhavat) [inline libs/dh/shared/data-access-msw/src/lib/mocks/data/marketParticipantUserRolePermissions.json:3]: nitpick: according to `SelectablePermissionsDto`, the `id` value should be a number

**Review by mimse — CHANGES_REQUESTED** [review]
Nice work 💯 

Just one thing

File: libs/dh/admin/data-access-api/src/lib/dh-admin-create-user-role-management-data-access-api.store.ts [hunk 3]
```diff
@@ -57,16 +54,16 @@ export class DhAdminCreateUserRoleManagementDataAccessApiStore extends Component
...
-        switchMap(() =>
+        switchMap((eicFunction) =>
           this.httpClientUserRole
-            .v1MarketParticipantUserRolePermissionsGet()
+            .v1MarketParticipantUserRolePermissionsGet(eicFunction)
             .pipe(
               tapResponse(
                 (response) => {
-                  this.updatePermissions(response);
+                  this.updatePermissions(response ?? []);
```
Comment (dzhavat) [inline libs/dh/admin/data-access-api/src/lib/dh-admin-create-user-role-management-data-access-api.store.ts:66]: question: can the `response` here be other than an array?

File: libs/dh/admin/data-access-api/src/lib/dh-admin-create-user-role-management-data-access-api.store.ts (hunk 1 above)
Comment (FirestarJes) [reply libs/dh/admin/data-access-api/src/lib/dh-admin-create-user-role-management-data-access-api.store.ts]: Fixed

File: libs/dh/shared/data-access-msw/src/lib/mocks/marketParticipantUserRole.ts [hunk 4]
```diff
@@ -36,7 +36,14 @@ function Permissions(apiBase: string) {
   return rest.get(
     `${apiBase}/v1/MarketParticipantUserRole/Permissions`,
     (req, res, ctx) => {
-      return res(ctx.json(marketParticipantUserRolePermissions));
+      const eicFunction = req.url.searchParams.get('eicFunction');
+      return res(
+        ctx.json(
+          marketParticipantUserRolePermissions.filter(
+            (s) => s.eicFunction == eicFunction
+          )
```
Comment (dzhavat) [inline libs/dh/shared/data-access-msw/src/lib/mocks/marketParticipantUserRole.ts:44]: praise: nice way to make the response look dynamic 👍

File: libs/dh/admin/feature-user-management/src/lib/feature-create-userrole/tabs/dh-create-userrole-tabs.component.html [hunk 5]
```diff
@@ -43,7 +44,8 @@ <h4>{{ t("headline") }}</h4>
         <watt-button
           type="submit"
           [disabled]="
-            this.userRole.permissions.length <= 0 || createRoleForm.invalid
+            this.userRole.permissions.length <= 0 ||
```
Comment (michaelmajgaard) [inline libs/dh/admin/feature-user-management/src/lib/feature-create-userrole/tabs/dh-create-userrole-tabs.component.html:47]: `=== 0` instead, right?

File: libs/dh/admin/data-access-api/src/lib/dh-admin-create-user-role-management-data-access-api.store.ts (hunk 3 above)
Comment (FirestarJes) [reply libs/dh/admin/data-access-api/src/lib/dh-admin-create-user-role-management-data-access-api.store.ts:66]: It shouldn't ever be, but in testing it did return null, so until i have had time to investigate i'd like to leave this in.

File: libs/dh/shared/data-access-msw/src/lib/mocks/data/marketParticipantUserRolePermissions.json (hunk 2 above)
Comment (FirestarJes) [reply libs/dh/shared/data-access-msw/src/lib/mocks/data/marketParticipantUserRolePermissions.json:3]: Fixed

File: libs/dh/admin/feature-user-management/src/lib/feature-create-userrole/tabs/dh-create-userrole-tabs.component.html (hunk 5 above)
Comment (FirestarJes) [reply libs/dh/admin/feature-user-management/src/lib/feature-create-userrole/tabs/dh-create-userrole-tabs.component.html]: Would also work, so fixed.

**Review by dzhavat — APPROVED** [review]

**Review by mimse — APPROVED** [review]

**Pull request merged**